server/controllers/applicant/getApplicantById.js

```diff
@@ -0,0 +1,9 @@
+const Applicant = require('../../database/models/applicant');
+
+const getApplicantById = (req, res) => {
+  Applicant.find({ _id: req.params.id })
+    .then((applicant) => res.json(applicant))
```

Comment:
@ha77an I think find() will return an array.  It might be clearer to the frontend if you just return one record, instead of that array (since there will only ever be one record for a given id).  Also, if there is no user for that id, find() will not throw an error, but will just return an empty array.  So your catch() won't handle that case and you might need some additional error handling:
```
Applicant.find({ ....
  .then((results) => {
    if (results.length) {
      res.json({data: results[0]});
    }
    else {
      res.status(404).json({error: 'user does not exist'});
    }
  })
  .catch(() => res.status(500).json({error: 'server error'});

Reply:
Or alternatively you could use `findById` which would always give you just one result.
https://mongoosejs.com/docs/api.html#model_Model.findById